src/SavingsDai.sol

```diff
@@ -309,9 +315,13 @@ contract SavingsDai {
     }
 
     function mint(uint256 shares, address receiver) external returns (uint256 assets) {
+        assets = mint(shares, receiver, 0);
+    }
+
+    function mint(uint256 shares, address receiver, uint16 referral) public returns (uint256 assets) {
         uint256 chi = (block.timestamp > pot.rho()) ? pot.drip() : pot.chi();
         assets = _divup(shares * chi, RAY);
-        _mint(assets, shares, receiver);
+        _mint(assets, shares, receiver, referral);
```

Comment:
We can change to emit the `Referral` events in this level (at the end of the new `mint()` and `deposit()` functions), and thus removing the requierement on `referral` being non 0.
It also removes the if condition in `_mint()` and saves a bit of gas.